### PR TITLE
Documents should be according to the BCF-API specification

### PR DESCRIPTION
**Pull request by jasollien**

### CODE REVIEW

File: Schemas/project.xsd [hunk 1]
```diff
@@ -12,7 +12,20 @@
...
 		<xs:sequence>
 			<xs:element name="Name" type="xs:string" minOccurs="0"/>
+			<xs:element name="Documents" minOccurs="0" maxOccurs="unbounded">
+				 <xs:complexType>
+					<!-- Guid of the document -->
+					<xs:attribute name="Guid" type="Guid" use="required" />
+					<!-- File name of the document -->
+					<xs:attribute name="FileName" type="xs:string" use="required" />
+				</xs:complexType>
+			</xs:element>
```
Comment (ykulbak) [inline Schemas/project.xsd:22]: Should there be a Document subelement (in a collection)?

**Comment by jasollien** [comment]
@ykulbak: Can you have a look at this? I also moved the shared type "Guid" to a seperate file so that it don't have to be repeated. Tested and fully working

**Comment by jasollien** [comment]
Example generated files:
Documents
```
//
// This file was generated by the JavaTM Architecture for XML Binding(JAXB) Reference Implementation, v2.3.2 
// See <a href="https://javaee.github.io/jaxb-v2/">https://javaee.github.io/jaxb-v2/</a> 
// Any modifications to this file will be lost upon recompilation of the source schema. 
// Generated on: 2020.11.04 at 11:49:03 AM CET 
//


package no.catenda.bimsync.bcf.schema.xml.v2_2;

import java.util.ArrayList;
import java.util.List;
import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;
import javax.xml.bind.annotation.XmlType;


/**
 * <p>Java class for anonymous complex type.
 * 
 * <p>The following schema fragment specifies the expected content contained within this class.
 * 
 * <pre>
 * &lt;complexType&gt;
 *   &lt;complexContent&gt;
 *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType"&gt;
 *       &lt;sequence&gt;
 *         &lt;element name="Document" type="{}Document" maxOccurs="unbounded" minOccurs="0"/&gt;
 *       &lt;/sequence&gt;
 *     &lt;/restriction&gt;
 *   &lt;/complexContent&gt;
 * &lt;/complexType&gt;
 * </pre>
 * 
 * 
 */
@XmlAccessorType(XmlAccessType.FIELD)
@XmlType(name = "", propOrder = {
    "document"
})
@XmlRootElement(name = "Documents")
public class Documents {

    @XmlElement(name = "Document")
    protected List<Document> document;

    /**
     * Gets the value of the document property.
     * 
     * <p>
     * This accessor method returns a reference to the live list,
     * not a snapshot. Therefore any modification you make to the
     * returned list will be present inside the JAXB object.
     * This is why there is not a <CODE>set</CODE> method for the document property.
     * 
     * <p>
     * For example, to add a new item, do as follows:
     * <pre>
     *    getDocument().add(newItem);
     * </pre>
     * 
     * 
     * <p>
     * Objects of the following type(s) are allowed in the list
     * {@link Document }
     * 
     * 
     */
    public List<Document> getDocument() {
        if (document == null) {
            document = new ArrayList<Document>();
        }
        return this.document;
    }

}
```
Document:
```
//
// This file was generated by the JavaTM Architecture for XML Binding(JAXB) Reference Implementation, v2.3.2 
// See <a href="https://javaee.github.io/jaxb-v2/">https://javaee.github.io/jaxb-v2/</a> 
// Any modifications to this file will be lost upon recompilation of the source schema. 
// Generated on: 2020.11.04 at 11:49:03 AM CET 
//


package no.catenda.bimsync.bcf.schema.xml.v2_2;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlType;


/**
 * <p>Java class for Document complex type.
 * 
 * <p>The following schema fragment specifies the expected content contained within this class.
 * 
 * <pre>
 * &lt;complexType name="Document"&gt;
 *   &lt;complexContent&gt;
 *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType"&gt;
 *       &lt;sequence&gt;
 *         &lt;element name="Guid" type="{}Guid"/&gt;
 *         &lt;element name="Filename" type="{http://www.w3.org/2001/XMLSchema}string"/&gt;
 *       &lt;/sequence&gt;
 *     &lt;/restriction&gt;
 *   &lt;/complexContent&gt;
 * &lt;/complexType&gt;
 * </pre>
 * 
 * 
 */
@XmlAccessorType(XmlAccessType.FIELD)
@XmlType(name = "Document", propOrder = {
    "guid",
    "filename"
})
public class Document {

    @XmlElement(name = "Guid", required = true)
    protected String guid;
    @XmlElement(name = "Filename", required = true)
    protected String filename;

    /**
     * Gets the value of the guid property.
     * 
     * @return
     *     possible object is
     *     {@link String }
     *     
     */
    public String getGuid() {
        return guid;
    }

    /**
     * Sets the value of the guid property.
     * 
     * @param value
     *     allowed object is
     *     {@link String }
     *     
     */
    public void setGuid(String value) {
        this.guid = value;
    }

    /**
     * Gets the value of the filename property.
     * 
     * @return
     *     possible object is
     *     {@link String }
     *     
     */
    public String getFilename() {
        return filename;
    }

    /**
     * Sets the value of the filename property.
     * 
     * @param value
     *     allowed object is
     *     {@link String }
     *     
     */
    public void setFilename(String value) {
        this.filename = value;
    }

}

```

File: Schemas/guid.xsd [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- With XMLSpy v2011 rel. 2 sp1 (http://www.altova.com) created by Jon Anders Sollien (Catenda AS) -->
```
Comment (jasollien) [inline Schemas/guid.xsd:2]: Should we remove these comments from all of our xsd's? I vote yes

Comment (ykulbak) [reply Schemas/guid.xsd]: Yes please

**Review by cschladetsch — COMMENTED** [review]
Agreed that plurals should be used for collections.

**Review by ykulbak — APPROVED** [review]

File: Schemas/markup.xsd [hunk 3]
```diff
@@ -96,11 +85,22 @@
...
+            <xs:choice>
+                <!-- Guid of the document: If pointing to an internal document -->
+                <xs:element name="DocumentGuid" type="Guid" minOccurs="0"/>
+                <!-- Url of the reference. If pointing to an external document -->
+                <xs:element name="Url" type="xs:string" minOccurs="0"/>
+            </xs:choice>
+            <!-- Human readable description of the document reference -->
+            <xs:element name="Description" type="xs:string" minOccurs="0"/>
+        </xs:sequence>
+        <xs:attributeGroup ref="DocumentReferenceAttributes"/>
```
Comment (pasi-paasiala) [inline Schemas/markup.xsd:99]: Could this be replaced with just guid? Like DocumentReferenceGuid?

**Review by pasi-paasiala — APPROVED** [review]
Looks good to me!

**Pull request merged**